sdk/CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+cmake_minimum_required(VERSION 3.22)
+
+include($ENV{FP_SDK_PATH}/cmake/riscv-toolchain.cmake)
+
+project(fp-sdk
+  DESCRIPTION "Software development kit (SDK) for FlexPRET"
+  LANGUAGES C ASM
+  VERSION 0.0.1
+)
+
+if (NOT DEFINED TARGET)
+  set(TARGET "emulator" CACHE STRING "Whether to compile SDK to target FlexPRET on emulator or FPGA" FORCE)
+endif()
+
+set(ENVS
+  FP_PATH
+  FP_SDK_PATH
+)
+
+foreach(env ${ENVS})
+  if (NOT DEFINED ENV{${env}})
+    message(FATAL_ERROR "Environment variable ${env} not set. To fix this\
+    step into the top-level directory and `source env.bash` or `source env.fish`.")
+  endif()
+endforeach()
+
+set(FP_INSTALLED_FILES
+  "flexpret/fp-emu"
+  "flexpret/hwconfig.cmake"
+  "flexpret/hwconfighash.txt"
+  "lib/include/flexpret/internal/hwconfig.h"
+  "lib/linker/internal/hwconfig.ld"
+)
+
+list(TRANSFORM FP_INSTALLED_FILES PREPEND "$ENV{FP_SDK_PATH}/")
+foreach(f ${FP_INSTALLED_FILES})
+  if (NOT EXISTS ${f})
+    message(FATAL_ERROR "Could not find ${f}. Please build FlexPRET and install\
+    it to the SDK with `cmake --install` to continue.")
+  endif()
+endforeach()
+
+include($ENV{FP_SDK_PATH}/flexpret/hwconfig.cmake)
+include($ENV{FP_PATH}/cmake/confighash.cmake)
+calculate_crc32()
+
+# Make it possible to override which configuration file to use from command line
+# Like so: cmake -DCMAKE_CONFIG=my_config --build .
+set(CONFIGFILE "$ENV{FP_SDK_PATH}/cmake/configs/default.cmake" CACHE PATH "Which configuration file to select")
+
+include(${CONFIGFILE})
+include($ENV{FP_SDK_PATH}/cmake/configverify.cmake)
+include($ENV{FP_SDK_PATH}/cmake/configderive.cmake)
+
+configure_file(
+  "$ENV{FP_SDK_PATH}/cmake/infiles/swconfig.h.in"
+  "$ENV{FP_SDK_PATH}/lib/include/flexpret/internal/swconfig.h"
+)
+
+configure_file(
+  "$ENV{FP_SDK_PATH}/cmake/infiles/swconfig.ld.in"
+  "$ENV{FP_SDK_PATH}/lib/linker/internal/swconfig.ld"
+)
+
+# Interface library
+add_library(fp-sdk-if INTERFACE)
```

Comment:
The `fp-sdk-if` library just has the compiler flags, header includes, linker script settings and so on. The `fp-sdk` has `fp-sdk-if` + source code. I had the issue that all the source code would be compiled multiple times when I linked it PRIVATE into different parts of the LF reactor-c parts. E.g.,

```
# Here I just want the header files, but `fp-sdk` compiles the source code as well
target_link_libraries(lf-low-level-platform PRIVATE fp-sdk)

# Here I actually want the source code
target_link_libraties(${LF_MAIN_TARGET} PRIVATE fp-sdk)
```

But it might be that I just misunderstood how this works. I'll try again and see if I can remove it - would be nice to only have `fp-sdk`.

Reply:
Hm, if it is possible, we only link `fp-sdk` with `lf-low-level-platform`. In this way we contain all platform-specific things to the `lf-low-level-platform` target. I believe this worked for `zephyr`, but I might be mistaken, the build logic for LF+reactor-c is not exactly trivial

Reply:
Hmm, it looks like removing `fp-sdk-if` leads to the issue I described. And I don't quite see an immediate solution to fix it without going away from the "source code interface" model we wanted (i.e., each app compiles the library for itself).

Reply:
I dont understand how merging them into a single target would make it not possible for an app to compile the library for itself?

It would, by the way, also be nice if the tests could be compiled standalone. E.g. I want to just compile the `add` test, but `cd tests/c-tests/add && cmake -Bbuild` does not work since it depends on being include from another CMakeLists.txt. It is good if they are standalone since that is a reference for the users for how to create their own apps.